pahfit/apfitter.py

```diff
@@ -0,0 +1,420 @@
+            multiplied with the components with multiplicative == True.
+
+        kwargs : dict
+            Arguments for the astropy model constructor, including a
+            unique value for "name". Should be generated with
+            self._astropy_model_kwargs; the register_() functions show how
+            to do this for each type of feature.
+
+        """
+        if multiplicative:
+            self.multiplicative_components.append(astropy_model_class(**kwargs))
+        else:
+            self.additive_components.append(astropy_model_class(**kwargs))
+
+    def register_starlight(self, name, temperature, tau):
+        """Register a BlackBody1D.
+
+        Parameters
+        ----------
+        name : str
+            Unique name.
+
+        temperature : array of size 3
+            Temperature for the blackbody, given as [value, lower_bound,
+            upper_bound]. The bounds assume the same system as the
+            features table, where masked == fixed, and +inf or -inf mean
+            unbounded.
+
+        tau : analogous. Used as amplitude.
+
+        """
+        self.component_types[name] = "starlight"
+        kwargs = self._astropy_model_kwargs(
+            name, ["temperature", "amplitude"], [temperature, tau]
+        )
+        self._register_component(BlackBody1D, **kwargs)
+
+    def register_dust_continuum(self, name, temperature, tau):
+        """Register a ModifiedBlackBody1D.
+
+        Analogous. Temperature and tau are used as temperature and
+        amplitude
+
+        """
+        self.component_types[name] = "dust_continuum"
+        kwargs = self._astropy_model_kwargs(
+            name, ["temperature", "amplitude"], [temperature, tau]
+        )
+        self._register_component(ModifiedBlackBody1D, **kwargs)
+
+    def register_line(self, name, power, wavelength, fwhm):
+        """Register a PowerGaussian1D
+
+        Analogous. Uses an implementation of the Gaussian profile, that
+        directly fits the power based on the internal PAHFIT units.
+
+        """
+        self.component_types[name] = "line"
+
+        kwargs = self._astropy_model_kwargs(
+            name,
+            ["amplitude", "mean", "stddev"],
+            [power, wavelength, np.array([x for x in fwhm]) / 2.355],
+        )
+        self._register_component(Gaussian1D, **kwargs)
+
+    def register_dust_feature(self, name, power, wavelength, fwhm):
+        """Register a PowerDrude1D.
+
+        Analogous. Uses an implementation of the Drude profile that
+        directly fits the power based on the internal PAHFIT units.
+
+        """
+        self.component_types[name] = "dust_feature"
+        kwargs = self._astropy_model_kwargs(
+            name, ["amplitude", "x_0", "fwhm"], [power, wavelength, fwhm]
+        )
+        self._register_component(Drude1D, **kwargs)
+
+    def register_attenuation(self, name, tau):
+        """Register the S07 attenuation component.
```

Comment:
We may add other attenuation models someday, in fact have a `model` parameter in the attenuation feature just for that.

Reply:
Yes, those are there waiting.  So you propose to leave those ignored in the API for now?  I suppose that's OK.  We will certainly have the problem when only one of the `Fitter`s implements something.  I suppose we can update the other to accept the argument and give an error or something.  Or we need a "Capabilities" flag setup that can be queried (`SupportAltAttenuationModels`, `SupportAttGeometry` etc.).  BTW, mixed vs. screen is in the original PAHFIT model, so should probably be supported OOTB.